From 413955540528b66702bf47e4e4d916240d22f43c Mon Sep 17 00:00:00 2001
From: Alexander Batischev <eual.jp@gmail.com>
Date: Sat, 29 Jun 2019 17:11:50 +0300
Subject: [PATCH] Do not fail libnewsboat build if git hash couldn't be
 obtained

Cirrus CI apparently deletes .git directory before the build. We want
our build script to work even without .git, though, so let's not call
`unwrap()` there.
---
 rust/libnewsboat/build.rs | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/rust/libnewsboat/build.rs b/rust/libnewsboat/build.rs
index d92fa35b..bb8d80a9 100644
--- a/rust/libnewsboat/build.rs
+++ b/rust/libnewsboat/build.rs
@@ -2,13 +2,13 @@ use std::process::Command;
 
 fn main() {
     // Code lifted from https://stackoverflow.com/a/44407625/2350060
-    // Panics in build script stop the build, which is fine by us - let's use unwrap() freely here.
-    let hash_output = Command::new("git")
+    if let Ok(hash_output) = Command::new("git")
         .args(&["describe", "--abbrev=4", "--dirty", "--always", "--tags"])
         .output()
-        .unwrap();
-    let hash = String::from_utf8_lossy(&hash_output.stdout);
-    println!("cargo:rustc-env=GIT_HASH={}", hash);
-    // Re-build this crate when Git HEAD changes. Idea lifted from vergen crate.
-    println!("cargo:rebuild-if-changed=.git/HEAD");
+    {
+        let hash = String::from_utf8_lossy(&hash_output.stdout);
+        println!("cargo:rustc-env=GIT_HASH={}", hash);
+        // Re-build this crate when Git HEAD changes. Idea lifted from vergen crate.
+        println!("cargo:rebuild-if-changed=.git/HEAD");
+    }
 }
